packages/language-service/ivy/language_service.ts

```diff
@@ -105,6 +119,50 @@ export class LanguageService {
     return results;
   }
 
+  private withCompiler<T>(fileName: string, p: (compiler: NgCompiler) => T): T {
+    const compiler = this.compilerFactory.getOrCreateWithChangedFile(fileName);
+    const result = p(compiler);
+    this.compilerFactory.registerLastKnownProgram();
+    return result;
+  }
+
+  getTcb(fileName: string, offset: number): GetTcbResponse {
```

Comment:
```suggestion
  getTcb(fileName: string, position: number): GetTcbResponse {
```
Suggestion: for consistency, let's use `position` here